lua/plenary/path.lua

```diff
@@ -463,10 +464,33 @@ function Path:copy(opts)
       opts.destination:sub(4, #opts.destination),
     }
   end
-
   local dest = Path:new(opts.destination)
 
-  return uv.fs_copyfile(self:absolute(), dest:absolute(), { excl = true })
+  if not self:is_dir() then
+    return uv.fs_copyfile(self:absolute(), dest:absolute(), { excl = true })
+  end
+  if opts.recursive then
+    dest:mkdir {
+      parents = F.if_nil(opts.parents, true, opts.parents),
+      exists_ok = F.if_nil(opts.exists_ok, true, opts.exists),
+    }
+    local data = {}
+    local scan = require "plenary.scandir"
+    scan.scan_dir(self.filename, {
+      respect_gitignore = F.if_nil(opts.respect_gitignore, true, opts.respect_gitignore),
+      hidden = F.if_nil(opts.hidden, false, opts.hidden),
+      depth = 1,
+      add_dirs = F.if_nil(opts.add_dirs, true, opts.add_dirs),
+      on_insert = function(entry)
+        table.insert(data, Path:new(entry))
```

Comment:
why insert `Path:new` in a table. Why not use the return of scan_dir and then create new Path items in the for loop below?